cmd/minikube/cmd/start.go

```diff
@@ -113,6 +115,8 @@ func runStart(cmd *cobra.Command, args []string) {
 	libMachineClient := libmachine.NewClient(constants.Minipath, constants.MakeMiniPath("certs"))
 	defer libMachineClient.Close()
 
+	setProxy()
```

Comment:
Instead of this _setProxy_ here, I rather split the content of this method into two parts, one which affects Docker and one which affects OpenShift. something like: _determineDockerEnv_ and _determineOpenShiftEnv_. The former can then be used within the cluster.MachineConfig setup.